README.md

```diff
@@ -1,2 +1,159 @@
 # ode-output-validator-library
+
 Contains shared library functions for validating ODE message output schemas and constraints.
+
+The shared library manages all validation checks and constraints are defined in a configuration ini file.
+
+## Installation
+
+TODO - Analyze value of pip support
+
+```
+pip install odevalidator
+```
+
+Once you have the package installed, import the TestCase class.
+
+```
+from odevalidator import TestCase
+```
+
+## Functional Interface
+
+### `TestCase(**kwargs)`
+
+Creates a configured test case object that can be used for validation.
+
+**Request Syntax**
+
+```
+test_case = TestCase(
+  filepath='string'  
+)
+```
+
+**Parameters**
+
+- **filepath** (_string_) \[REQUIRED\] Relative or absolute path to configuration file (see more information in the configuration section below).
+
+**Return Type**
+
+`Object`
+
+**Usage Example**
+```
+test_case = TestCase("./config/bsmLogDuringEvent.ini")
+```
+
+### `.validate(**kwargs)`
+
+Iterates a message queue and performs validations on each message, returning the results.
+
+**Request Syntax**
+
+```
+results = test_case.validate(
+  msg_queue=queue.Queue()
+)
+```
+
+**Parameters**
+
+- **msg_queue** (_queue.Queue_) \[REQUIRED\] A [`queue`](https://docs.python.org/3/library/queue.html) containing messages to be validated.
```

Comment:
What's the reason for using a Queue rather than passing the list to the `validate` method?

Reply:
The main advantage of using Queue over a list type is thread safety. [In the validation script](https://github.com/usdot-jpo-ode/jpo-ode/blob/master/validation-script/test-harness.py#L52) I create a list of messages received by forking a Kafka listener thread and then passing the queue as a parameter.

I don't think foresee any thread use internally to the library, so this  may be overkill. If you think its easier, we can just use a normal list.

Reply:
Great! I think it's absolutely a good idea to foresee a need to parallel processing of the data. You never know in what context this library will be used.